### DISCUSSION TITLE
TERMUX PIP VERSION

### DISCUSSION THREAD

**Discussion by jaymielayo** [termux/termux-app#2881] [Q&A]
HELLO I JUST START BUT I GOTTA PROBLEM WITH THIS PIP VERSION CAN SOMEBODY HELP ME?

IF I USE A PIP IT GET ME THIS WARNING 
```
WARNING: You are using pip version 19.2.3, however version 20.3.4 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

THEN IF I CHECK MY PIP VERSION
```
pip --version
pip 22.1.2 from /data/data/com.termux/files/usr/lib/python3.10/site-packages/pip (python 3.10)
```

Helpp! 😦😭

**Comment by 2096779623** [comment]
`pip install --upgrade pip`

You should check out the tips first, then google it instead of asking someone else.

**Reply by jaymielayo** [reply]
`pip install --upgrade pip
Requirement already satisfied: pip in /data/data/com.termux/files/usr/lib/python3.10/site-packages (22.1.2)`

It doesn't work i surfed the internet for answer i try deleting and installing it again doesnt work too

**Reply by achu-sudo** [reply]
pip install --upgrade pip
ERROR: Installing pip is forbidden, this will break the python-pip package (termux).
😭😭😭

**Reply by achu-sudo** [reply]
Pls help

**Reply by sylirre** [reply]
@achu-sudo Just don't run `pip install --upgrade pip`, this is no longer supported by Termux and provided version is up-to-date.

If something suggests you to update pip version, then know this is just a some dumb-troubleshooting step provided unconditionally to filter out people filing false bug reports for a package. It has nothing to do with reality and is displayed always regardless of reason of actual failure.

When module installation failed, please read the full error log and try to understand the issue, adjust your installation steps accordingly. You maybe missing some dependency or pip flag. Thus for example some modules may require packages `rust`, `cmake` or `ninja` to be installed and `--no-build-isolation` to be passes for `pip install`.

Termux pip now may be only upgraded through `pkg upgrade` and only if we release a new version. Pay attention that we may change package logic at our discretion, if we think this is necessary. For example as pip is now provided through python-pip package, we have disabled the self-upgrade feature.

**Reply by sylirre** [reply]
In addition to above: on deprecated Python v2.x the pip no longer can be upgraded. Although t will always prompt to update the pip because looks for the same URL for updates, obsolete Python release no longer receive compatible module versions.

`apt purge python2`

**Comment by sylirre** [comment]
Make sure you are not using `pip` from Python v2.x. This one shows that warning, but can't be upgraded since Python v2 is deprecated.

**Reply by jaymielayo** [reply]
Wdym?
